tests/certification/state/cockroachdb/cockroachdb_test.go

```diff
@@ -0,0 +1,282 @@
+	goclient "github.com/dapr/go-sdk/client"
+	"github.com/dapr/kit/logger"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	sidecarNamePrefix       = "cockroach-sidecar-"
+	dockerComposeYAML       = "docker-compose.yml"
+	stateStoreName          = "statestore"
+	certificationTestPrefix = "stable-certification-"
+	stateStoreNoConfigError = "error saving state: rpc error: code = FailedPrecondition desc = state store is not configured"
+)
+
+func TestCockroach(t *testing.T) {
+	log := logger.NewLogger("dapr.components")
+
+	stateStore := state_cockroach.New(log).(*state_cockroach.CockroachDB)
+	ports, err := dapr_testing.GetFreePorts(3)
+	assert.NoError(t, err)
+
+	stateRegistry := state_loader.NewRegistry()
+	stateRegistry.Logger = log
+	stateRegistry.RegisterComponent(func(l logger.Logger) state.Store {
+		return stateStore
+	}, "cockroachdb")
+
+	currentGrpcPort := ports[0]
+	currentHTTPPort := ports[1]
+	// appPort := 34842
+
+	// Generate a unique value for the key being inserted to ensure no conflicts occur
+	keyOne := uuid.New()
+	keyOneString := strings.Replace(keyOne.String(), "-", "", -1)
+
+	basicTest := func(ctx flow.Context) error {
+		client, err := goclient.NewClientWithPort(fmt.Sprint(currentGrpcPort))
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+
+		err = stateStore.Ping()
+		assert.Equal(t, nil, err)
+
+		err = client.SaveState(ctx, stateStoreName, certificationTestPrefix+"key1", []byte("certificationdata"), nil)
+		assert.NoError(t, err)
+
+		// get state
+		item, err := client.GetState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		assert.NoError(t, err)
+		assert.Equal(t, "certificationdata", string(item.Value))
+
+		errUpdate := client.SaveState(ctx, stateStoreName, certificationTestPrefix+"key1", []byte("cockroachCertUpdate"), nil)
+		assert.NoError(t, errUpdate)
+		item, errUpdatedGet := client.GetState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		assert.NoError(t, errUpdatedGet)
+		assert.Equal(t, "cockroachCertUpdate", string(item.Value))
+
+		// delete state
+		err = client.DeleteState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		assert.NoError(t, err)
+
+		// get state
+		item, errUpdatedGet = client.GetState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		assert.NoError(t, errUpdatedGet)
+		assert.Equal(t, []byte(nil), item.Value)
+		return nil
+	}
+
+	// Check if TCP port is actually open
+	testGetAfterCockroachdbRestart := func(ctx flow.Context) error {
+		client, err := goclient.NewClientWithPort(fmt.Sprint(currentGrpcPort))
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+
+		err = stateStore.Ping()
+		assert.Equal(t, nil, err)
+
+		resp, err := stateStore.Get(&state.GetRequest{
+			Key: keyOneString,
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, "2", *resp.ETag) //2 is returned since the previous etag value of "1" was incremented by 1 when the update occurred
```

Comment:
There need to be spaces after comments.

please lint this file (`make lint` will not run cover this file automatically)